arch/arm/src/stm32h7/stm32_tickless.c

```diff
@@ -0,0 +1,1057 @@
+   */
+
+  if (g_tickless.base == STM32_TIM6_BASE ||
+      g_tickless.base == STM32_TIM7_BASE)
+    {
+      return -EINVAL;
+    }
+
+  /* Frozen mode because we don't want to change the GPIO, preload register
+   * disabled.
+   */
+
+  ccmr_val = (GTIM_CCMR_MODE_FRZN << GTIM_CCMR1_OC1M_SHIFT);
+
+  /* Set polarity */
+
+  ccer_val |= GTIM_CCER_CC1P << GTIM_CCER_CCXBASE(channel);
+
+  /* Define its position (shift) and get register offset */
+
+  if ((channel & 1) != 0)
+    {
+      ccmr_val  <<= 8;
+      ccmr_mask <<= 8;
+    }
+
+  if (channel > 1)
+    {
+      ccmr_offset = STM32_GTIM_CCMR2_OFFSET;
+    }
+
+  ccmr_orig  = stm32_getreg16(ccmr_offset);
+  ccmr_orig &= ~ccmr_mask;
+  ccmr_orig |= ccmr_val;
+  stm32_putreg16(ccmr_offset, ccmr_orig);
+  stm32_putreg16(STM32_GTIM_CCER_OFFSET, ccer_val);
+
+  return OK;
+}
+
+/****************************************************************************
+ * Name: stm32_interval_handler
+ *
+ * Description:
+ *   Called when the timer counter matches the compare register
+ *
+ * Input Parameters:
+ *   None
+ *
+ * Returned Value:
+ *   None
+ *
+ * Assumptions:
+ *   Called early in the initialization sequence before any special
+ *   concurrency protections are required.
+ *
+ ****************************************************************************/
+
+static void stm32_interval_handler(void)
+{
+#ifdef CONFIG_SCHED_TICKLESS_ALARM
+  struct timespec tv;
+#endif
+  tmrinfo("Expired...\n");
+
+  /* Disable the compare interrupt now. */
+
+  stm32_tickless_disableint(g_tickless.channel);
+  stm32_tickless_ackint(g_tickless.channel);
+
+  g_tickless.pending = false;
+
+#ifndef CONFIG_SCHED_TICKLESS_ALARM
+  nxsched_timer_expiration();
+#else
+  up_timer_gettime(&tv);
+  nxsched_alarm_expiration(&tv);
+#endif
+}
+
+/****************************************************************************
+ * Name: stm32_timing_handler
+ *
+ * Description:
+ *   Timer interrupt callback.  When the freerun timer counter overflows,
+ *   this interrupt will occur.  We will just increment an overflow count.
+ *
+ * Input Parameters:
+ *   None
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+static void stm32_timing_handler(void)
+{
+  g_tickless.overflow++;
+
+  STM32_TIM_ACKINT(g_tickless.tch, GTIM_SR_UIF);
+}
+
+/****************************************************************************
+ * Name: stm32_tickless_handler
+ *
+ * Description:
+ *   Generic interrupt handler for this timer.  It checks the source of the
+ *   interrupt and fires the appropriate handler.
+ *
+ * Input Parameters:
+ *   None
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+static int stm32_tickless_handler(int irq, void *context, void *arg)
+{
+  int interrupt_flags = stm32_tickless_getint();
+
+  if (interrupt_flags & GTIM_SR_UIF)
```

Comment:
```suggestion
  if ((interrupt_flags & GTIM_SR_UIF) != 0)
```

Reply:
I'm not going to do these.  I've reached my point of it being too nitpicky. Are we really going through every instance of this logic in the codebase and changing it to `!= 0`.  Is it really any more readable?

Reply:
It is not about readability. It is about that I do not have time to open each and every define in the code to check if is a bit position or mask or a combination of those and explicit compare again value make me sure what author meant. If `GTIM_SR_UIF` is a bit, then I expect to see `if ((interrupt_flags & GTIM_SR_UIF) != 0)` or `if ((interrupt_flags & GTIM_SR_UIF) == GTIM_SR_UIF)`. If `GTIM_SR_UIF` is mask then with `if ((interrupt_flags & GTIM_SR_UIF) != 0)` I'm sure that author really means that any of the set bits in mask leads to a true condition or `if ((interrupt_flags & GTIM_SR_UIF) == GTIM_SR_WHATEVER)` means that only `GTIM_SR_WHATEVER` bits are expected to be set. Please understand me right, I just want to be sure that changes that are submitted are not just a copy-pasted code with possible bugs. Your answer that this change is intentional fully satisfies me. Thank you for answering.

Reply:
@pkarashchenko I think I understand your logic a bit more now.  Thanks for explaining.